File: WordGames/novowels/novowels.py
# Program to create a list for all of the words with no vowel and not even a Y

# importing the word list
sow_pods = open("/home/evergreen/UnderDogsAssignments/sowpods")

# Go through each word in the file and add it to the list, words
words = []
novowels = []

for line in sow_pods:
    words.append(line.strip())

for w in words:
    if ("A" not in w) and ("E" not in w) and ("I" not in w) and ("O" not in w)and ("U" not in w) and ("Y" not in w):
        novowels.append(w)

print('All of the words that have no vowels, including Y are ' + str(novowels))

# Find the longest word  in the lost of novowels
longest = ''
length = len(longest)
for y in novowels:
    if len(y) > length:
        longest = y
        length = len(longest)

print('The longest word with no vowels in it is ' + longest)

File: WordGames/novowels/noVowels.js
// Program to find all the words with no vowels
// Importing the fs module
let fs = require("fs")

// Intitializing the readFileLines with the file
const readFileLines = filename =>
   fs.readFileSync(filename)
   .toString('UTF8')
   .split('\n');

// Calling the readFiles function with file name
let words = readFileLines('/home/evergreen/UnderDogsAssignments/sowpods');
noVowels = []

// Cycle through words and append words that do not have AEIO or U in them to noVowels
for(let i = 0; i < words.length; i++) {
    if (!words[i].includes("A") && !words[i].includes("E") && !words[i].includes("I") && !words[i].includes("O") && !words[i].includes("U") && !words[i].includes("Y"))
        noVowels.push(words[i]);}


console.log("The words containing no vowels are " + noVowels)

//setting variables to compare index in noVowels
let longestNoVowels = "SH";
for (let j = 0; j < noVowels.length; j++) {
    if (noVowels[j].length > longestNoVowels.length) {
        longestNoVowels = noVowels[j];
    } else {
        continue
    }}
console.log("The longest word with no vowels is " + longestNoVowels)